address/address_v1.go

```diff
@@ -67,6 +82,25 @@ func (v *v1) FromHex(s string) (Address, error) {
 }
 
 func (v *v1) decodeBech32(encodedAddr string) ([]byte, error) {
+	if len(encodedAddr) != V1AddressStringLength {
+		return nil, errors.Wrapf(ErrInvalidAddr, "address length = %d, expecting 41", len(encodedAddr))
+	}
+	hrp, grouped, err := bech32.Decode(encodedAddr)
+	if err != nil {
+		return nil, errors.Wrapf(ErrInvalidAddr, err.Error())
+	}
+	if hrp != prefix() {
+		return nil, errors.Wrapf(ErrInvalidAddr, "hrp %s and address prefix %s don't match", hrp, prefix())
+	}
+	// Group the payload into 8 bit groups.
+	payload, err := bech32.ConvertBits(grouped, 5, 8, false)
+	if err != nil {
+		return nil, errors.Wrapf(ErrInvalidAddr, err.Error())
+	}
+	return payload, nil
+}
+
+func (v *v1) decodeBech32Legacy(encodedAddr string) ([]byte, error) {
 	hrp, grouped, err := bech32.Decode(encodedAddr)
 	if hrp != prefix() {
 		return nil, errors.Wrapf(err, "hrp %s and address prefix %s don't match", hrp, prefix())
```

Comment:
here `err` is actually nil, `errors.Wrapf(err)` is still nil, causing error escape